tests/chainer_tests/test_variable.py

```diff
@@ -983,358 +977,320 @@ def v():
-            assert gx.dtype == x_var.grad.dtype
-            assert x_var.grad.device is expected_device
-            np.testing.assert_array_equal(
-                backend.CpuDevice().send(x_var.grad),
-                backend.CpuDevice().send(gx))
-            assert x_var.grad_var is not None
-            with pytest.raises(RuntimeError):
-                x_var.grad_var.node
+    def to_device_method(self, device, allow_unchain):
+        # Device is `None` or a `GpuDevice`.
+        if device is not None:
+            assert device.xp is cuda.cupy
+            device = device.device.id
+        self.x.to_gpu(device, allow_unchain=allow_unchain)
+
+    def check_to_device_method(self, src_device, dst_device):
+        assert isinstance(dst_device, backend.GpuDevice)
+        x = self.x
+
+        assert x.xp is cuda.cupy
+        assert x.node is not None
+        assert x.device == dst_device
+        assert isinstance(x.array, cuda.cupy.ndarray)
+        assert x.array.device.id == dst_device.device.id
+
+        if self.x_with_grad:
+            assert x.grad_var.xp is cuda.cupy
+            assert x.grad_var.node is not None
+            assert x.grad_var.device == dst_device
+            assert isinstance(x.grad, cuda.cupy.ndarray)
+            assert x.grad.device.id == dst_device.device.id
         else:
-            assert x_var.grad is None
-            assert x_var.grad_var is None
+            assert x.grad is None
+            assert x.grad_var is None
+
+        no_copy = (
+            src_device.xp is cuda.cupy
+            and src_device.device.id == dst_device.device.id
+            and not self.x_with_graph)
+        if no_copy:
+            assert x.array is self.x_src
+            if self.x_with_grad:
+                assert x.grad is self.gx_src
+
+    def test_to_device_method_allow_unchain_none_device(
+            self, src_backend_config, dst_backend_config):
+        self.setup(src_backend_config)
 
-        assert x_var.xp is chainerx
-        assert x_var._has_chainerx_array is True
+        self.to_device_method(None, True)
 
-    def test_to_chx_from_numpy(self):
-        self.check_to_chx(self.x, self.gx)
+        self.check_to_device_method(
+            src_backend_config.device, backend.GpuDevice.from_device_id(
+                cuda.cupy.cuda.runtime.getDevice()))
+        self.check_to_device_method_ndarrays()
 
-    @attr.gpu
-    def test_to_chx_from_cupy(self):
-        self.check_to_chx(cuda.to_gpu(self.x), cuda.to_gpu(self.gx))
 
-    # TODO(hvy): Write test when implemented.
-    @attr.ideep
-    def test_ideep_to_chx(self):
-        raise unittest.SkipTest('Not yet supported')
+@testing.parameterize(*testing.product({
+    'x_shape': [(10,), ()],
+    'x_with_grad': [True, False],
+    'x_with_graph': [True, False],
+}))
+@testing.backend.inject_backend_tests(None, _numpy_backend_params)
+# TODO(hvy): Test intel64.
+@testing.backend.inject_backend_tests(None, _backend_params)
```

Comment:
Hmm, how about adding a TODO at the definition of `_backend_params` instead?
These lines themselves do not seem to need fixes.

Reply:
Is there any reason intel64 cannot be tested for now?

Reply:
The reason is `to_chx`, which won't work on intel64 ndarray holding variables (`RuntimeError`). It is probably possible for the other methods though.

Reply:
OK, please use skip.

Reply:
Needed some changes/skips to the tests but fixed to also test intel64.